### PR TITLE
dpdk: fix DPDK crash on startup v2

### PR DESCRIPTION
**Pull request by lukashino**
Redmine ticket: https://redmine.openinfosecfoundation.org/issues/6877
Describe changes:
-  updated the DPDK versions,
- replaced "custom" Suricata worker sync to a pthread_barrier,
- static variables changed to also thread_local, 
- fixed the redmine ticket - a linked list of thread slots assigned the current slot instead of the next one,
- added new SW tests to GitHub CI, runs Suricata with predefined configs to make sure Suricata can run.

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 19771

File: src/runmode-dpdk.c [hunk 1]
```diff
@@ -1596,8 +1596,7 @@ static void *ParseDpdkConfigAndConfigureDevice(const char *iface)
     if (iconf->workers_sync == NULL) {
```
Comment (victorjulien) [inline src/runmode-dpdk.c:1596]: interesting, I didn't know about pthread_barrier. The commit needs a lot more explanation of what it does.

File: src/source-dpdk.h [hunk 2]
```diff
@@ -77,7 +72,7 @@ typedef struct DPDKIfaceConfig_ {
     /* threads bind queue id one by one */
     SC_ATOMIC_DECLARE(uint16_t, queue_id);
     SC_ATOMIC_DECLARE(uint16_t, inconsistent_numa_cnt);
-    DPDKWorkerSync *workers_sync;
+    pthread_barrier_t *workers_sync;
```
Comment (victorjulien) [inline src/source-dpdk.h:75]: add comment on purpose here

File: .github/workflows/dpdk/suricata-null-ids.yaml [hunk 3]
```diff
@@ -0,0 +1,2174 @@
+%YAML 1.1
```
Comment (victorjulien) [inline .github/workflows/dpdk/suricata-null-ids.yaml:1]: we should be able to do this w/o a full copy of the yaml

**Review by victorjulien — CHANGES_REQUESTED** [review]

File: src/runmode-dpdk.c (hunk 1 above)
Comment (lukashino) [reply src/runmode-dpdk.c:1596]: ok, btw FYI it is not available for MacOS but I believe it fits the DPDK scenario.

**Comment by lukashino** [comment]
Follow-up in https://github.com/OISF/suricata/pull/10723

**Pull request closed without merging**